Automattic/wp-calypso
# Add debugging to Calypso Sentry library

#### Proposed Changes

This is modifications to the `client/lib/sentry` library extracted from https://github.com/Automattic/wp-calypso/pull/63634 so that it can support manual calls to `captureException()` from error boundaries.

All of this props @noahtallen 

_From @noahtallen:_
At this point, the main thing this adds to Sentry is debug information. To support, that, I also added the Sentry types package and bumped the Sentry versions. I slightly clarified how Sentry is enabled. This doesn't meaningfully change how Sentry is used in production. Most of the changes happened as I was debugging a Sentry error which was not getting reported, so I was tweaking code here and there to see if anything would fix it. While it didn't get fixed, I think adding the debugger is still useful. 

#### Testing Instructions
- Load the calypso.live image
- Execute `localStorage.setItem( 'debug', 'calypso:sentry*' )` in the browser console
- Reload the page. Navigate around a bit and see if any debug statements show up
